### PR TITLE
Sitewise

### PR DESCRIPTION
**Pull request by maximcherny**
*Issue #, if available:* n/a

*Description of changes:* created a new SIteWise accelerator for ingestion of data from an OPC UA-compatible endpoint to be streamed into AWS IoT SiteWise. This accelerator is to form basis for two of the existing workshops, namely IIoT Security and IoT analytics. The accelerator follows the pre-existing pattern used to create other V2 accelerators, namely `etl_simple` and `os_cmd`.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,7 +15,9 @@ The current V2 accelerators are:
...
   - This accelerator deploys additional functionality on to the _Base Implementation_ stack to demonstrate sending an operating system command as an MQTT message from AWS IoT Core, running the command locally, and publishing back the command output to another MQTT topic in the cloud.
 - [Extract, Transform, and Load  - Simple Implementation](v2/etl_simple)
-  - This accelerator deploys additional functionality on to the _Base Implementation_ stack to demonstrate how to extract, transform, and load (ETL) data from a local device to the cloud. This accelerator breaks the three steps into discrete processes, decoupled by using [Interprocess Communcation](https://docs.aws.amazon.com/greengrass/v2/developerguide/interprocess-communication.html). 
+  - This accelerator deploys additional functionality on to the _Base Implementation_ stack to demonstrate how to extract, transform, and load (ETL) data from a local device to the cloud. This accelerator breaks the three steps into discrete processes, decoupled by using [Interprocess Communcation](https://docs.aws.amazon.com/greengrass/v2/developerguide/interprocess-communication.html).
+- [SiteWise Deployment](v2/sitewise)
```
Comment (gadams999) [inline README.md:19]: AWS IoT SiteWise to be consistent with service naming.

File: v2/README.md [hunk 2]
```diff
@@ -14,3 +14,5 @@ Once the _base_ has been deployed and tested, you can then select one of the oth
   - This accelerator deploys additional functionality on to the _Base Implementation_ stack to demonstrate sending an operating system command as an MQTT message from AWS IoT Core, running the command locally, and publishing back the command output to another MQTT topic in the cloud.
 - [Extract, Transform, and Load - Simple Implementation](etl_simple)
   - This accelerator deploys additional functionality on to the _Base Implementation_ stack to demonstrate how to extract, transform, and load (ETL) data from a local device to the cloud. This accelerator breaks the three steps into discrete processes, decoupled by using [Interprocess Communication](https://docs.aws.amazon.com/greengrass/v2/developerguide/interprocess-communication.html).
+- [SiteWise Deployment](sitewise)
```
Comment (gadams999) [inline v2/README.md:17]: AWS IoT SiteWise

File: v2/sitewise/README.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
+# AWS IoT Greengrass V2 SiteWise Deployment
```
Comment (gadams999) [inline v2/sitewise/README.md:1]: AWS IoT SiteWise

File: v2/sitewise/README.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+This deployment, SiteWise Deployment, extends the capabilities of the [base](../base) accelerator stack by adding a new thing group, and Greengrass deployment for the existing Greengrass core. The functionality of the Greengrass deployment leverages several AWS-provided components with the ability to ingest data from an OPC UA endpoint using a SiteWise gateway component and stream the data to IoT SiteWise.
```
Comment (gadams999) [inline v2/sitewise/README.md:3]: AWS IoT SiteWise

File: v2/sitewise/README.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+
+- Thing group - A group specific to this accelerator, the existing thing is added and targeted for the new component
+- IoT policy - An AWS IoT policy attached to the existing certificate extending the ability to publish batch property values to IoT SiteWise
+- Greengrass deployment - The Greengrass components targeted for the new Thing group.
+
+Once this is fully deployed, the Greengrass core device will receive and deploy the required components immediately or when next started.
+
+> **NOTE:** This accelerator is intended for educational use only and should not be used as the basis for production workloads without fully reviewing the component and it's artifacts.
+
+## SiteWise Deployment Use Case
```
Comment (gadams999) [inline v2/sitewise/README.md:15]: AWS IoT SiteWise

File: v2/sitewise/README.md [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+- [AWS IoT SiteWise Workshop](https://iot-sitewise.workshop.aws/en/introduction.html)
+- [AWS IoT SiteWise Edge Workshop](https://iot-sitewise-edge.workshop.aws/)
+- [IIoT Security Workshop](https://catalog.us-east-1.prod.workshops.aws/workshops/5b543f4c-1952-4bd9-96c8-b009c16da2bc/en-US)
+
+## Design Pattern
+
+The following architecture shows the deployment of this accelerator (aligned to the base implementation).
+
+![SiteWise Deployment Architecture](docs/arch.svg)
```
Comment (gadams999) [inline v2/sitewise/README.md:31]: AWS IoT SiteWise

File: v2/sitewise/README.md [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+
+1. The CDK stack creates a new thing group (`sitewise-group`), adds the existing `thing-core`, and deploys the required components. The two thing group deployments are merged and sent to the Greengrass core.
+1. The OPC UA collector component starts and ingests data from the configured OPC UA endpoint.
+1. The data are sent to the SiteWise edge publisher component via the local stream manager.
+1. The SiteWise edge publisher component publishes the data to AWS IoT SiteWise, via the newly created SiteWise gateway associated with the Greengrass deployment.
```
Comment (gadams999) [inline v2/sitewise/README.md:36]: IoT SiteWise publisher

File: v2/sitewise/README.md [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+
+## Folder Structure
+
+```text
+sitewise
```
Comment (gadams999) [inline v2/sitewise/README.md:41]: Suggest changing to `aws_iot_sitewise`

File: v2/sitewise/README.md [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+
+1. Review any CloudWatch Logs log groups and delete these if needed.
+
+All traces of the component including the thing group and additional AWS IoT Core policy permissions have been removed from the Greengrass core implementation.
+
+### Frequently Asked Questions
+
+#### Should I use this code for my production workloads?
+
+In its current form, no. This component is not suitable for production workloads and intended to demonstrate the capabilities of Greengrass for data ingestion using OPC UA. To be production-ready would require support for additional configuration options to support various autehentication mechanisms to cater for different OPC UA endpoint configurations.
```
Comment (gadams999) [inline v2/sitewise/README.md:137]: AWS IoT Greengrass.

File: v2/sitewise/cdk/bin/app.ts [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+const app = new cdk.App()
+const name = process.env.STACK_NAME || "gg-accel-stitewise"
+
+new SiteWiseStack(app, name, {
+  env: {
+    account: process.env.CDK_DEPLOY_ACCOUNT || process.env.CDK_DEFAULT_ACCOUNT,
+    region: process.env.CDK_DEPLOY_REGION || process.env.CDK_DEFAULT_REGION
+  },
+  stackName: name,
+  description: "Greengrass Accelerator v2 - SiteWise Component"
```
Comment (gadams999) [inline v2/sitewise/cdk/bin/app.ts:17]: Greengrass Accelerator v2 - AWS IoT SiteWise Component

I think it's fine to use shorted names in the code, but for any descriptive output or reference, such as stack name, full service name is more consistent.

**Review by gadams999 — CHANGES_REQUESTED** [review]
If you could make the doc and folder layout, I'll then exercise the accelerators.

**Comment by maximcherny** [comment]
@gadams999 thanks for reviewing, I've gone through and made the suggested changes.

**Pull request merged**